internal/incident/incident.go

```diff
@@ -119,74 +109,47 @@ func (i *Incident) ProcessEvent(ctx context.Context, tx *sqlx.Tx, ev event.Event
 	return i.notifyContacts(ctx, tx, &ev, causedBy)
 }
 
-func (i *Incident) processSeverityChangedEvent(ctx context.Context, tx *sqlx.Tx, ev event.Event) (types.Int, error) {
-	oldIncidentSeverity := i.Severity()
-	oldSourceSeverity := i.SeverityBySource[ev.SourceId]
-	if oldSourceSeverity == event.SeverityNone {
-		oldSourceSeverity = event.SeverityOK
+func (i *Incident) processSeverityChangedEvent(ctx context.Context, tx *sqlx.Tx, ev event.Event, created bool) (types.Int, error) {
+	oldIncidentSeverity := i.Severity
+	if oldIncidentSeverity == event.SeverityNone {
+		oldIncidentSeverity = event.SeverityOK
 	}
 
-	if oldSourceSeverity == ev.Severity {
+	var causedByHistoryId types.Int
+	newIncidentSeverity := ev.Severity
+	if oldIncidentSeverity == newIncidentSeverity && !created {
```

Comment:
What is the `&& !created` you added for? If the incident is created, the old severity should be ok and the new severity something else, so that should already be captured by the other part of the condition.

Reply:
If the incident has just been created, it will first enter `Incident#processIncidentOpenedEvent()` and set the `severity` to the event severity.
https://github.com/Icinga/icinga-notifications/blob/288093e269564567e3d50966310adc02be107591/internal/incident/incident.go#L186-L188
Afterwards, when it goes in here (`Incident#processSeverityChangedEvent()`), old and new severities are the same, of course.

Reply:
I see, looks like you're changing more than I expected: Previously, incidents were always created in an ok state, so the first event opening the incident wasn't treated differently from later events in this function. You're changing this to directly creating the incident at the right severity. Nothing wrong with that, but this should probably go with some other code changes then, for example, is there any case left where `processSeverityChangedEvent()` would actually do anything if called with `create=true`? So if you change it so that there is no longer an explicit severity change when opening the incident, this function probably shouldn't even be called when creating an incident.

You can probably also see this in the generated history, with the current main branch, it looks something like that:

1. Opened this incident
2. Source Icinga 2 reported a severity change from Ok to Warning: [...output/message...]
3. Incident severity changed from Ok to Warning

What I'd have expected from this PR would be a history like:

1. Opened this incident
2. Severity changed from Ok to Warning: [...output/message...]

But with the changes you started, it should probably look something like this:

1. Opened incident at severity Warning: [...output/message...]

Which probably is more on point, but as I said, required slightly more extensive changes, which might also explain why you were confused about the cause column in the history table.